src/main/java/dao/LocalDroneDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package main.java.dao;
+
+import main.java.api.dtos.Drone;
+import main.java.api.dtos.DroneType;
+import main.java.entity.DroneEntity;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LocalDroneDaoImpl implements LocalDroneDao {
+
+    private static final String DRONE_FILE_NAME = "drone_data.bin";
+    private static List<DroneEntity> singletonList = new ArrayList<>();
+
+    /**
+     * Initialize or Update Drone Information
+     * @param drones from API
+     */
+    public void updateDroneData(List<DroneEntity> drones) {
+        saveDroneData(drones);
+        getDroneDataFromFile();
+    }
+
+    private void saveDroneData(List<DroneEntity> drones) {
+        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(DRONE_FILE_NAME))) {
+            oos.writeObject(drones);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Get cached drone data from file or heap memory
+     * @return List of all drone data
+     */
+    public List<DroneEntity> loadDroneData() {
```

Comment:
Please clarify in the documentation, if applicable, that the singleton list is intended to be initialized only once.